07.ls_object/lib/ls_command.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require 'pathname'
+require_relative 'ls_file'
+
+class LsCommand
+  COLS = 3
+
+  def show_files
+    options = parse_options
+    ls_files = build_ls_files(options)
+    if options[:long]
+      show_ls_files_in_long_format(ls_files)
+    else
+      show_ls_files_in_default_format(ls_files)
+    end
+  end
+
+  private
+
+  def parse_options
+    options = { all: false, reverse: false, long: false }
+
+    OptionParser.new do |opts|
+      opts.on('-a') { options[:all] = true }
+      opts.on('-r') { options[:reverse] = true }
+      opts.on('-l') { options[:long] = true }
+    end.parse!
+
+    options
+  end
+
+  def build_ls_files(options)
+    target_directory = Pathname(ARGV[0] || '.')
+    basenames = Dir.foreach(target_directory)
+    ls_files = basenames.map { |basename| LsFile.new(basename, target_directory) }
+    filtered_ls_files = options[:all] ? ls_files : ls_files.reject(&:hidden?)
+    options[:reverse] ? filtered_ls_files.sort.reverse : filtered_ls_files.sort
+  end
+
+  def show_ls_files_in_long_format(ls_files)
+    puts "total #{ls_files.sum(&:block_size)}"
+    max_length = find_max_lengths_by_props(ls_files)
```

Comment:
```suggestion
    max_lengths = find_max_lengths_by_props(ls_files)
```

左辺と右辺の名前を揃えた方がいいかも
https://bootcamp.fjord.jp/pages/readable-variable-name